### ISSUE TITLE
[Request] Option to have post-repaired changes to ships, NPCs, and Fleets (personalities and governments)

### ISSUE THREAD

**Issue by Zitchas** [endless-sky/endless-sky#7127]
### Problem Description

Currently, when a ship is repaired from a disabled state, it almost invariably throws itself back into battle according to its defined personality(ies), often resulting in it being destroyed. Sometimes it might hang out in the fringes of the battle until it's recovered some shields, but that doesn't seem to happen very often.

### Related Issue Links

Didn't see any.

### Desired Solution

For example, the human storyline makes a big deal of the Oathkeepers; a branch of the Navy that are basically people who have been defeated in battle against the Free Worlds, and released on their oath that they won't take part in the war.

What I'd like to see is being able to do something like this: In a battle against the Navy in any of the relevant missions, I would like to be able to disable a navy ship, then board to repair it. When I do so, the ship is repaired as normal, but it switches government to the Oathkeeper and gains the pacifist personality, and in either case immediately flees the system and is considered defeated by the mission parameters.

Basically, I'm looking for options to adjust the behaviors of ships, NPCs, and fleets after they've been repaired so that they behave differently afterwards. This would mostly revolve around changing their government and/or personalities, but giving them navigational commands (ex. Leave the system in the direction of the nearest planet friendly to them, or land if there's a friendly planet in system).

This is our current fleet definition:
```
fleet <name>
	government <name>
	names <phrase>
	fighters <phrase>
	cargo <cargo#>
	commodities <commodity>...
	outfitters <outfitter>...
	personality <type>...
		<type>...
		confusion <amount#>
	variant [<weight#>]
		<model> [<count#>]
		...
```

I'd like to see something like this added to it as an optional thing:
```
fleet <name>
	...
	repair <hostile/friendly>
		government <name>
		personality <type>
			<type>
			confusion <amount#>
		...
```

So there'd be three repair optional things:
- repair friendly
- repair hostile
- repair

`repair`: This would specify different what changes with the ship after it has been repaired as a general thing (for instance, a timid merchant might become a pacifist and flee battle no matter who repaired them).
`repair friendly`: This specifies any changes if the ship is repaired by an ally or other friendly ship. (for example, navy units that are repaired by their allies might become heroic and disabling, since their enemy was merciful and didn't destroy them outright)
`repair hostile`: This specifies any changes if the ship is repaired by an enemy or other hostile ship. (for example, if the player in the FW campaign repairs a Navy ship that they disabled, it might become an "Oathkeeper" ship on the spot, and gain the pacifist personality.

Importantly, when a hostile ship is repaired by the player, there should be a way for the mission to recognize that the ship has been defeated and is no longer relevant. Ideally, this would vary based on the mission. Some missions may require that the target ship is *destroyed*, and obviously just defeating and repairing it won't suffice. (disabling and capturing probably shouldn't either, but that's another matter.) Other missions just need the ships to be defeated and changed governments. Some, such as the Navy/Oathkeeper situation, might be satisfied by the disable and repair into being an oathkeeper, others might not accept that and specify that it needs to be destroyed or captured. 

### Alternative Approaches

-

### Additional Context

-

**Comment by timegrinder** [comment]
Love this idea, would it also be able to handle temporarily affecting the personality of player-captured ships, or recently repaired escorts? 

Too often a ship will be captured, and while still crippled, dive head-first into an ongoing or future fight. It could instead, if the player chose, act as an unarmed ship and attempt to avoid combat until it had landed at a spaceport for the first time (or if it has hull repair, once it gets back up to a decent level).

I often capture ships in the heat of a battle (while the main firefight is away from the vessel being boarded) and constantly have to deal with that vessel then hiking it full speed toward its destruction, so it would be nice to be able to handle that differently.

I understand though if that becomes a 'Player should micro-manage that better' or other plausible behaviour coding options for the future.

**Comment by Zitchas** [comment]
The main problem with enforcing this on player shots is that we don't know how the player wants them to act. Since players may prefer their shots always retreat and heal, for others they may be simply Canon fodder that they want to plunge back into the fray to distract the AI. As such, I would prefer to keep this focused on usage for NPCs and fleets, whose behavior is established by story and lore.

**Comment by timegrinder** [comment]
The idea was to have it as a player configurable option such as a button on the boarding window that is available after capture  (as people may want some ships passive, while others to fight and a global setting wouldn't fit) which is why I understand it might be out of scope for this and to be treated as a separate feature entirely.
 
As for the implementation you've discussed: Would this make sense to be set up for occurrences outside of the specific mission tree you've mentioned?

While I'm not 100% on lore, does it make sense for the possibility of having other groups that have a peaceful  + hostile pair have ships with a very small percentage chance to join their peaceful counterpart?

As you've said with Navy turning Oathkeeper, there may be the rare Unfettered Hai who are tired of fighting and swaps after being 'rescued' by their enemy, or (extremely rare) Korath Exiles turning Efreti, or possibly Pirates on the odd occasion turning neutral.

Each group would have their different % chance of turning peaceful depending on their original government, assumedly with Navy having a significantly higher chance when repaired by FW during the war.

**Comment by Zitchas** [comment]
Definitely. Actually the biggest place that I want such functionality is with fleets defined in governments, and particular those called by system definitions.

And I have use cases I would be implementing in Remnant systems immediately.

**Comment by Hurleveur** [comment]
With the navy behavior you describe it would be impossible to acquire navy ships during the conflict (and its currently impossible at other times)

Also currently its impossible to repair a hostile ships. Changing that could already be a PR; and thus issue, on its own (trough another hotkey for instance, 'r', or maybe some ships have a special tag that makes them be repaired even if hostile) it would make sense. But thats an issue on its own, according to me.

**Comment by Zitchas** [comment]
> With the navy behavior you describe it would be impossible to acquire navy ships during the conflict (and its currently impossible at other times)

Which part? Allies already repair, and this wouldn't be forcing the player to repair, merely giving them the option to do so. It would, of course, severely cut down on the number of disabled shops floating around if enemies are routinely boarding, accepting their oaths, and sending them on their way. So player captures would have to be more intentional and specific target than opportunistic and general.

**Comment by Hurleveur** [comment]
Currently we cant repair hostile ships, I assumed you intended for this to default on repairing and not give option to board.

**Comment by Zitchas** [comment]
My intent wasn't to change the default behavior. So if it is a friendly ship, the default is to repair. If it is hostile, then to board. But yes, that's a good point that there would also need to be a modifier or something to change what the action is. Perhaps shift+boarding key? I see two options:

1. Standardize it so that the regular boarding key always repairs, and shift-board always boards. (or the other way, just that it doesn't change, so the player can internalize one keypress always does the same thing)
2. Stick with the current default pattern: If friendly, then board repairs the target. If hostile, then board boards the target. Shift-board flips the action to the other option.
3. Change the boarding screen to be standardized, so when we board we get the usual looting panel and capture side panel; but now have an additional button, perhaps as a panel across the bottom, that reads "Repair ship."

I'd personally prefer option 1 or 3, as I like to be in command. Option 2 is basically reducing the player to saying "Do something to this ship" and the game picking "board or repair" based on hostility status. I dislike issuing unknown commands, or having the AI making decisions for me.

Option 3 might actually be the best, as it enables the player doing what I'd like to see the Remnant doing: Board the ship, loot the weapons and valuables off it, then repair it and send it on its way.

**Comment by Zern9528** [comment]
I like option 3.
And it should effect your rep too.
And it should definitely be chance whether or not they "convert", like what timegrinder suggested.

**Comment by UnorderedSigh** [comment]
I was able to implement this by changing the government and personality after an event (board or disable). I ran into two problems so far:

1. Fighters are no longer the same government as their carrier.
2. If a ship changes to a friendly government when the player has targeted it and is shooting at it, then the new government (and its allies) will become hostile.

**Comment by TheGiraffe3** [comment]
Can this happen sooner rather than later?
